### PR TITLE
feat(2199): Enhance metrics with event downtime information [2]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Need to add event downtime data to metrics function.

## Objective

This PR enhances the metrics method with event downtime information.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2199

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -46,6 +49,8 @@ const eventMetrics = async event => {
     let totalQueuedTime = queuedTime || 0;
     let totalImagePullTime = imagePullTime || 0;
     const builds = [{ id, jobId, duration, status }];
+    let isDowntimeEvent =
+        ((downtimeJobs.length === 0 || downtimeJobs.includes(jobId)) && downtimeStatuses.includes(status)) || false;
```
Comment (pritamstyz4ever) [inline lib/pipeline.js]: do we need the || false?? It should evaluate to boolean right?

File: lib/pipeline.js [hunk 2]
```diff
@@ -74,14 +79,21 @@ const eventMetrics = async event => {
             duration,
             status
         });
+
+        // Set downtime event flag if job is in downtimeJobs and status is in downtimeStatuses
+        if ((downtimeJobs.length === 0 || downtimeJobs.includes(jobId)) && downtimeStatuses.includes(status)) {
+            isDowntimeEvent = true;
+        }
     }
```
Comment (pritamstyz4ever) [inline lib/pipeline.js:87]: Looks like the condition can be reused ?

File: lib/pipeline.js [hunk 3]
```diff
@@ -1544,6 +1573,26 @@ class PipelineModel extends BaseModel {
...
+                        dayjs(downtimeEvents[i].maxEndTime),
+                        'second'
+                    );
+                }
+
+                downtimeEvents[i].downtimeDuration = dayjs(new Date()).diff(
+                    dayjs(downtimeEvents[i].maxEndTime),
+                    'second'
+                );
+            }
```
Comment (pritamstyz4ever) [inline lib/pipeline.js:1594]: Probably use map fn?

File: test/lib/pipeline.test.js [hunk 4]
```diff
@@ -2791,13 +2791,14 @@ describe('Pipeline Model', () => {
         });
     });
 
-    describe('get metrics', () => {
+    describe.only('get metrics', () => {
```
Comment (jithine) [inline test/lib/pipeline.test.js]: remove `only`

File: lib/pipeline.js (hunk 2 above)
Comment (tkyi) [reply lib/pipeline.js:87]: the way it's written, the first element in buildMetrics is set manually, so I'm also setting `isDowntimeEvent` that way
do you have a better suggestion?

File: lib/pipeline.js (hunk 3 above)
Comment (tkyi) [reply lib/pipeline.js:1594]: need to do something different with last element in array
gets the duration based off current time vs next element's time

File: test/lib/pipeline.test.js (hunk 4 above)
Comment (tkyi) [reply test/lib/pipeline.test.js]: updated

File: lib/pipeline.js (hunk 1 above)
Comment (tkyi) [reply lib/pipeline.js]: removed

**Review by jithine — COMMENTED** [review]
need to fix tests, otherwise LGTM

**Review by jithine — APPROVED** [review]

**Pull request merged**